framework/PostProcessors/DataMining.py

```diff
@@ -213,159 +214,175 @@ def _localGenerateAssembler(self,initDict):
-            inputDict['Features'][hist][param] = currentInput._dataContainer['outputs'][hist][param]
-
-      inputDict['metadata'] = currentInput.getAllMetadata()
-
-    ## Redundant if-conditional preserved as a placeholder for potential future
-    ## development working directly with files
-    # elif isinstance(currentInp, Files.File):
-    #   self.raiseAnError(IOError, 'Unsupported input type (' + currentInput.subtype + ') for PostProcessor ' + self.name + ' must be a PointSet.')
+      inputDict['metadata'] = currentInput.getMeta(pointwise=True,general=True)
+      return inputDict
+
+    elif self.PreProcessor is not None:
+      return self.inputToInternalForPreProcessor(currentInput)
+
+  def inputToInternalForPreProcessor(self,currentInput):
+    """
+      Function to convert the received input into a format that this
+      post-processor can understand
+      @ In, currentInput, object, DataObject of currentInput
+      @ Out, inputDict, dict, an input dictionary that this post-processor can process
+    """
+    inputDict = {'Features': {}, 'parameters': {}, 'Labels': {}, 'metadata': {}}
+    if self.PreProcessor.interface.returnFormat('output') not in ['PointSet']:
+      self.raiseAnError(IOError, 'DataMining PP: this PP is employing a pre-processor PP which does not generates a PointSet.')
+
+    tempData = self.PreProcessor.interface.inputToInternal([currentInput])
+    preProcessedData = self.PreProcessor.interface.run(tempData)
+
+    if self.initializationOptionDict['KDD']['Features'] == 'input':
+      featureList = currentInput.getVars('input')
+    elif self.initializationOptionDict['KDD']['Features'] == 'output':
+      dataList = preProcessedData['data'].keys()
+      # FIXME: this fix is due to the changes in the data structure of interface pp
+      toRemove = ['prefix', 'ProbabilityWeight'] + currentInput.getVars('input')
+      featureList = [elem for elem in dataList if elem not in toRemove]
     else:
-      self.raiseAnError(IOError, 'Unsupported input type (' + currentInput.type + ') for PostProcessor ' + self.name + ' must be a PointSet.')
+      featureList = [feature.strip() for feature in self.initializationOptionDict['KDD']['Features'].split(',')]
+    for key in featureList:
+      inputDict['Features'][key] = copy.deepcopy(preProcessedData['data'][key])
+
+    inputDict['metadata'] = currentInput.getMeta(pointwise=True,general=True)
     return inputDict
 
+  def inputToInternal(self, currentInp):
+    """
+      Function to convert the received input into a format this object can
+      understand
+      @ In, currentInp, list or DataObjects, Some form of data object or list of
+        data objects handed to the post-processor
+      @ Out, inputDict, dict, An input dictionary this object can process
+    """
+
+    if type(currentInp) == list:
+      currentInput = currentInp[-1]
+      '''====> For the reviewer: should we raise a warning/error here? I see issues here......'''
```

Comment:
The list comes from the Step (The Step always send the Inputs  (```<Input>```) ) as a list. 
We might want to error out in case the number of inputs >1

Reply:
fixed.